src/core/chain.rs

```diff
@@ -198,16 +193,13 @@ mod tests {
             }};
         }
 
-        let result = assert_handle!(2, handler_continue, handler_stop, handler_error);
-        assert!(matches!(result, HandlerResult::Stop));
-
-        let result = assert_handle!(1, handler_stop, handler_continue, handler_error);
-        assert!(matches!(result, HandlerResult::Stop));
+        let result = assert_handle!(2, handler_ok, handler_error);
+        assert!(matches!(result, HandlerResult::Err(_)));
 
-        let result = assert_handle!(1, handler_error, handler_stop, handler_continue);
-        assert!(matches!(result, HandlerResult::Error(_)));
+        let result = assert_handle!(1, handler_error, handler_ok);
+        assert!(matches!(result, HandlerResult::Err(_)));
```

Comment:
```suggestion
        assert_eq!(result, HandlerResult::Err(_));
```
In case of error you will have more information about what happens.
This is not critical note

Reply:
You can not compare result since it does not implement corresponding traits

Reply:
Ah, boxed error